search-impl/src/test/java/com/example/auction/search/impl/SearchServiceImplTest.java

```diff
@@ -0,0 +1,273 @@
+package com.example.auction.search.impl;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.actor.ActorRef;
+import akka.stream.Materializer;
+import com.example.auction.bidding.api.*;
+import com.example.auction.item.api.*;
+import com.example.auction.search.api.SearchRequest;
+import com.example.auction.search.api.SearchResult;
+import com.example.auction.search.api.SearchService;
+import com.example.elasticsearch.ElasticsearchTestUtils;
+import com.lightbend.lagom.javadsl.api.ServiceCall;
+import com.lightbend.lagom.javadsl.api.ServiceLocator;
+import com.lightbend.lagom.javadsl.api.broker.Topic;
+import com.lightbend.lagom.javadsl.testkit.ServiceTest;
+import com.lightbend.lagom.javadsl.testkit.ServiceTest.Setup;
+import org.junit.*;
+import org.pcollections.PSequence;
+
+import javax.inject.Inject;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.bind;
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.defaultSetup;
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.eventually;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+
+import com.example.core.*;
+import scala.concurrent.duration.FiniteDuration;
+
+// This TestCase puts a lot of pressure on elasticserach destroying and rebuilding indices on each run. Currently all
+// tests are ignored but can be run on a freshly installed elasticserach 5.0.x individually.
+// This TestCase expects an elasticsearch 5.0.x available on port 9200.
+public class SearchServiceImplTest {
```

Comment:
What if the tests could be structured so that it doesn't require destroying the index between each test method?

One option that comes to mind is to include a bit of random text in the item name and the search query for each test method.

Alternatively, if the category search were fleshed out, you could use a random category for each item and include that in the search.